python_modules/dagster/dagster_tests/definitions_tests/decorators_tests/test_asset_check_decorator.py

```diff
@@ -799,3 +800,35 @@ def check1():
     assert check_eval.check_name == "check1"
 
     assert check_eval.target_materialization_data is None
+
+
+def test_direct_invocation():
+    @asset_check(asset="asset1")
+    def check1():
+        return AssetCheckResult(passed=True)
+
+    assert check1()
+
+
+def test_direct_invocation_with_context():
+    @asset_check(asset="asset1")
+    def check1(context):
+        return AssetCheckResult(passed=True)
+
+    assert check1(build_op_context())
```

Comment:
https://github.com/dagster-io/dagster/issues/21115

holding off since AssetCheckExecutionContext is currently controversial